### ISSUE TITLE
embedding lab TIFF to png with alpha channel fails

### ISSUE THREAD

**Issue by DaGaMs** [lovell/sharp#646]
I'm experiencing a bug with a specific type of tiff file (16bit, lab) when I try to convert it to an thumbnail in RGBA with transparent background. The error message is:

```
vips_colourspace: no known route from 'multiband' to 'lab'
vips_colourspace: no known route from 'multiband' to 'srgb'
```

Interestingly, when I set the `background` to `white` instead of `{r: 0, g: 0, b: 0, a: 0}`, the error does *not* occur, but the background is black instead of white!

I've looked through the previous discussion relating to this error, but couldn't find any clues as to what might be going on with this particular image (_i.e._ it has an ICC profile, and it's not a GIF)

Here's the original image:
![LAB TIFF](https://dl.dropboxusercontent.com/u/816576/tiff_16-bit-lab_example.tiff)

The code looks roughly like this:

```javascript
var resizer = sharp()
  .sequentialRead();

resizer = resizer.resize(width, height)
  .background({r: 0, g: 0, b: 0, a: 0})
  .embed();

resizer = resizer.png();

fileStream
  .pipe(resizer)
  .on("info", (info) => {
    console.log('Read file of size ' + info.size);
  })
  .on("error", (err) => {
    console.error("Error processing image data: " + err);
  })
  .pipe(outputStream);
```

Installed tool versions:

- sharp: 0.16.2
- vips: `libvips 8.5.0-Sat Nov 12 18:29:10 UTC 2016`
- GraphicsMagick: 1.3.23

Any help would be much appreciated!

**Comment by lovell** [comment]
Hello, this is the first attempt I'm aware of to process a (CIE)LAB TIFF image using sharp, so congratulations. I'll need to investigate how best to fix/support this.

Is this sample image you've provided either in the public domain or CC licensed as it would make an excellent addition to the test suite?

(As an aside, colourspace handling will be getting a bit of a makeover with #218.)

**Comment by DaGaMs** [comment]
This image is mine, I am happy to provide it under a CCBY. In fact, it's already public on Flickr somewhere under that license.

--
Benjamin Schuster-Böckler
ben@evershelfapp.com
http://twitter.com/evershelf
snap. search. share.

**Comment by lovell** [comment]
Commit d245526 allows the embed (and extend) operations to work with an alpha-transparency background using non-(s)RGB input images.

libvips was unable to determine the colourspace as being LAB+alpha approximately [here](https://github.com/jcupitt/libvips/blob/master/libvips/colour/colourspace.c#L526). Luckily we already know the colourspace so can pass it via the `source_space` parameter.

In addition, the RGBA colour passed to the `background()` function had to be converted into the LAB colourspace before the embed operation.

This will be fixed in v0.17.0 - thanks for reporting.

**Comment by lovell** [comment]
v0.17.0 now available.
